Fix: Stripe online payments being wrongfully marked as successful

Matthias-Ab opened the pull request:
## Description
During checkout with Stripe, if the user selected a payment method that would redirect them out of InvoicePlane (ex. Revolut Pay, Klarna, etc..) the user could then cancel the transaction by clicking the "Back" or "Cancel" buttons and InvoicePlane would mark the invoice as paid due to incorrect callback handling.

## Related Issue
- https://github.com/InvoicePlane/InvoicePlane/issues/1208

## Motivation and Context
Fixes the very important open issue listed above.

## Pull Request Checklist

  * [x] My code follows the code formatting guidelines.
  * [x] I have an issue ID for this pull request.
  * [x] I selected the corresponding branch.
  * [x] I have rebased my changes on top of the corresponding branch.

## Issue Type (Please check one or more)

  * [x] Bugfix
  * [ ] Improvement of an existing Feature
  * [ ] New Feature

nielsdrost7 commented on application/modules/guest/controllers/gateways/Stripe.php:
```diff
@@ -73,53 +73,102 @@ public function create_checkout_session($invoice_url_key)
-            $this->session->set_flashdata('alert_success', sprintf(trans('online_payment_payment_successful'), $invoice->invoice_number));
+        // 3) DEBUG LOGGING (optional but recommended)
+        //    Make sure your PHP error log is set up (log_errors=On in php.ini)
+        error_log("Stripe callback reached. Checkout session id: " . $checkout_session_id);
```
codeigniter has a function log_message() which makes the logging end up on `/application/logs/`:
```
log_message('debug', 'Logging about Stripe');
```

nielsdrost7 commented on application/modules/guest/controllers/gateways/Stripe.php:
```diff
@@ -73,53 +73,102 @@ public function create_checkout_session($invoice_url_key)
-                trans('online_payment_payment_failed') . '<br/>' . $$e->getMessage());
+                sprintf(trans('online_payment_payment_failed')) .
+                '<br/>' .
+                'Payment was not completed. Stripe session status: ' . $session->payment_status
```
use as much translations as you can, if a translation is missing, just add it in `/application/language/ip_lang.php`

nielsdrost7 commented on application/modules/guest/controllers/gateways/Stripe.php:
```diff
@@ -73,53 +73,102 @@ public function create_checkout_session($invoice_url_key)
+                'merchant_response_successful' => false,
+                'merchant_response_date'       => date('Y-m-d'),
+                'merchant_response_driver'     => 'stripe',
+                'merchant_response'            => 'Transaction not completed or canceled',
```
Translation?

nielsdrost7 commented:
@Matthias-Ab thanks for the PR, man!
It's looking good.

Would you mind making those tiny changes?

We're using the `log_message` function so it ends up in `/application/logs` instead of the PHP log.
And We're trying to offer as much translations as possible, so users can translate text in their own language using crowdin

Matthias-Ab commented:
@nielsdrost7 Thanks for the feedback! I've updated the code accordingly:

Replaced the error_log() calls with CodeIgniter’s log_message() so logs now correctly appear in /application/logs.
Updated all user-facing messages to use translation keys.
Added the necessary translation keys to ip_lang.php for full localization support.
Let me know if you need any further tweaks!

Matthias-Ab commented:
@idressos
